src/github_repo_request_local.py

```diff
@@ -0,0 +1,74 @@
+import subprocess
+from pathlib import Path
+import pandas as pd
+from loguru import logger
+import shutil
+
+
+def count_and_list_test_files(directory):
+    logger.info(f"Processing the directory {directory}")
+    test_files = []
+    # Define a set of file extensions to exclude
+    excluded_extensions = {
+        ".txt",
+        ".md",
+        ".h",
+        ".xml",
+        ".html",
+        ".json",
+        ".png",
+        ".jpg",
+    }
+    for path in directory.rglob("*"):
+        # Check if the path is a file and either the file or its parent directory contains 'test'
+        if path.is_file() and (
+            "test" in path.name.lower() or "test" in str(path.parent).lower()
+        ):
+            # Exclude files with certain extensions
+            if path.suffix not in excluded_extensions:
+                test_files.append(str(path))
+    return len(test_files), test_files
+
+
+updated_csv_path = Path("../data/updated_local_github_df_test_count.csv")
```

Comment:
If practical, please can you modify the code to use `git_codebase_root(...)` rather than a hard-coded path. This would make the code less dependent on the folder it's run from.

Reply:
- Added the function `git_codebase_root` to make the script less dependent on the folder it's   
run from and more versatile.

- It returns the absolute path of the top-level directory of the current Git repository. If
not in a Git repository, returns None.

- committed and pushed the code